src/group_theory/submonoid/operations.lean

```diff
@@ -766,6 +766,27 @@ lemma mrange_snd : (snd M N).mrange = ⊤ :=
 lemma mrange_inl_sup_mrange_inr : (inl M N).mrange ⊔ (inr M N).mrange = ⊤ :=
 by simp only [mrange_inl, mrange_inr, prod_bot_sup_bot_prod, top_prod_top]
 
+/-- The `mul_hom` from the comap of a submonoid to itself. -/
+@[to_additive "the `add_hom` from the comap of an additive submonoid to itself.", simps]
+def from_comap (f : M →* N) (N' : submonoid N) :
+  N'.comap f →* N' :=
+{ to_fun := λ x, ⟨f x, x.prop⟩,
+  map_one' := subtype.eq f.map_one,
+  map_mul' := λ x y, subtype.eq (f.map_mul x y) }
+
+/-- The `mul_hom` from a submonoid to its image. -/
+@[to_additive "the `add_hom` from an additive submonoid to its image.", simps]
+def to_map (f : M →* N) (M' : submonoid M) :
```

Comment:
Can you use this to prove `add_equiv.add_submonoid_equiv_map` further down the file?

Also based on that name, this should be called `monoid_hom.submonoid_map`

Reply:
Alternatively, maybe the other definition should be renamed.

Reply:
> Can you use this to prove `add_equiv.add_submonoid_equiv_map` further down the file?
> 

I could change the proof to 
```
{ to_fun := λ x, ⟨e x, _⟩,
  inv_fun := λ x, ⟨e.symm x, _⟩, -- we restate this for `simps` to avoid `⇑e.symm.to_equiv x`
  ..e.to_monoid_hom.submonoid_map S, ..equiv.image e.to_equiv S }
```
I'm not sure if this is any better though.

Reply:
`simps` should know better than to generate that. I guess it doesn't matter too much though. Can you reference the hom version from the equiv docstring?